apps/src/javalab/JavalabView.jsx

```diff
@@ -55,18 +58,27 @@ class JavalabView extends React.Component {
     setIsTesting: PropTypes.func,
     canRun: PropTypes.bool,
     canTest: PropTypes.bool,
-    showProjectTemplateWorkspaceIcon: PropTypes.bool.isRequired,
     longInstructions: PropTypes.string,
     hasContainedLevels: PropTypes.bool,
     awaitingContainedResponse: PropTypes.bool,
     isSubmittable: PropTypes.bool,
-    isSubmitted: PropTypes.bool
+    isSubmitted: PropTypes.bool,
+    setDisableFinishButton: PropTypes.func
   };
 
   componentDidMount() {
     this.props.onMount();
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.isReadOnlyWorkspace !== this.props.isReadOnlyWorkspace) {
+      const disableFinishButton =
```

Comment:
nit: can you add a comment here explaining why this logic means we should disable the finish button?

Reply:
Sounds good, I'll do a follow-up for this. I really just copied this from StudioApp, since it was dependent on readonly workspace, and that is variable now, this would also be variable.